docs/design/gh-ost-integration.md

```diff
@@ -0,0 +1,509 @@
+2022-03-07 09:49:42 INFO Tables renamed
+2022-03-07 09:49:42 INFO Lock & rename duration: 1.003875334s. During this time, queries on `person` were blocked
+[2022/03/07 09:49:42] [info] binlogsyncer.go:164 syncer is closing...
+[2022/03/07 09:49:42] [error] binlogstreamer.go:77 close sync with err: sync is been closing...
+[2022/03/07 09:49:42] [info] binlogsyncer.go:179 syncer is closed
+2022-03-07 09:49:42 INFO Closed streamer connection. err=<nil>
+2022-03-07 09:49:42 INFO Dropping table `db`.`_person_ghc`
+2022-03-07 09:49:42 INFO Table dropped
+2022-03-07 09:49:42 INFO Am not dropping old table because I want this operation to be as live as possible. If you insist I should do it, please add `--ok-to-drop-table` next time. But I prefer you do not. To drop the old table, issue:
+2022-03-07 09:49:42 INFO -- drop table `db`.`_person_del`
+2022-03-07 09:49:42 INFO Done migrating `db`.`person`
+2022-03-07 09:49:42 INFO Removing socket file: /tmp/gh-ost.db.person.sock
+2022-03-07 09:49:42 INFO Tearing down inspector
+2022-03-07 09:49:42 INFO Tearing down applier
+2022-03-07 09:49:42 INFO Tearing down streamer
+2022-03-07 09:49:42 INFO Tearing down throttler
+# Done
+```
+
+In the output, you can see the main components of gh-ost.
+
+**Migrator** is the main schema migration flow manager.
+
+**Inspector** reads data from the read-MySQL-server (typically a replica, but can be the master). It is used for gaining initial status and structure, and later also followup on progress and changelog.
+
+**Streamer** reads data from binary logs and streams it on.
+
+**Applier** connects and writes the applier-server, which is the server where migration happens. Applier is the one to actually write row data and apply binlog events onto ghost table.
+
+**Throttler** collects metrics related to throttling and makes informed decision whether throttling should take place. Here throttling means to pause.
+
+#### Requirements and limitations
+
+- `gh-ost` currently requires MySQL versions 5.6 and greater.
+- You will need to have one server serving Row Based Replication (RBR) format binary logs. Right now `FULL` row image is supported. `MINIMAL` to be supported in the near future. `gh-ost` prefers to work with replicas. You may still have your primary configured with Statement Based Replication (SBR).
+- If you are using a replica, the table must have an identical schema between the primary and replica.
+- `gh-ost` requires an account with these privileges:
+  - `ALTER, CREATE, DELETE, DROP, INDEX, INSERT, LOCK TABLES, SELECT, TRIGGER, UPDATE` on the database (schema) where your migrated table is, or of course on `*.*`
+  - either:
+    - `SUPER, REPLICATION SLAVE` on *.*, or:
+    - `REPLICATION CLIENT, REPLICATION SLAVE` on *.*
+The `SUPER` privilege is required for `STOP SLAVE, START SLAVE` operations. These are used on:
+- Switching your `binlog_format` to `ROW`, in the case where it is not ROW and you explicitly specified `--switch-to-rbr`
```

Comment:
We may want users to make the switch themselves but telling them how to do that and consequences - this is part of the task check. Otherwise, it might impact other workload using binlogs.

Reply:
We don't have the `SUPER` privilege and we won't switch binlog format for the user. 

> We may want users to make the switch themselves but telling them how to do that and consequences 

Yes. We will include that in the documentation.